src/features/SpellCheckerProvider.ts

```diff
@@ -202,8 +202,10 @@ export default class SpellCheckerProvider implements vscode.CodeActionProvider
 
 		text = this.processUserIgnoreRegex( text );
 
-		// remove pandoc yaml header
-		text = text.replace( /---(.|\n)*\.\.\./g, ' ' );
```

Comment:
This removes YAML frontmatter from Pandoc Markdown files (http://pandoc.org/MANUAL.html#extension-yaml_metadata_block) and should not effect all MD documents. If you have a sample document, I can take a look at it to see what's going on. I have no plan to remove this search from the main extension.